code/headers/test_usart.cpp

```diff
@@ -0,0 +1,41 @@
+#include "test_usart.hpp"
+
+namespace r2d2 {
+
+    test_usart_c::test_usart_c(unsigned int baudrate, usart_ports_c controller):
+        baudrate(baudrate),
+        controller(controller)
+    {}
+
+    void test_usart_c::enable(){
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::disable(){
+        //Not needed in test implementation
+    }
+
+    bool test_usart_c::send(const uint8_t c) {
+        //Not needed in test implementation
+    }
+
+    void test_usart_c::putc(char c) {
+        send(c);
+    }
+
+    uint8_t test_usart_c::receive() {
+        return 0xAA;
+    }
+
+    bool test_usart_c::char_available() {
+        return true;
+    }
+
+    char test_usart_c::getc() {
+        return receive();
+    }
+
+    unsigned int test_usart_c::available() {
+        return 1;
```

Comment:
why is this a unsigned int? not a size_t if it returns the amount of data available and bool if it only returns if data is available.

Reply:
because the base class returns a unsigned int